src/components/Utils/queries.js

```diff
@@ -23,31 +23,52 @@ const histogramQueryStrForEachField = (field) => {
   }`);
 };
 
-const queryGuppyForAggs = (path, type, fields, gqlFilter, acc) => {
+const cardinalityQueryStrForEachField = (field) => {
+  const splittedFieldArray = field.split('.');
+  const splittedField = splittedFieldArray.shift();
+
+  if (splittedFieldArray.length === 0) {
+    return (`
+      ${splittedField} {
+        _cardinalityCount
+      }
+      `);
+  }
+  return (`
+  ${splittedField} {
+    ${cardinalityQueryStrForEachField(splittedFieldArray.join('.'))}
+  }`);
+};
+
+const queryGuppyForAggs = (path, type, fields, countFields, gqlFilter, acc) => {
```

Comment:
```suggestion
const queryGuppyForAggs = (path, type, fields, cardinalityFields, gqlFilter, acc) => {
```

I think this renaming can better reflect the purpose of this field

Also I think we should make it defaults to `{}` to make it optional for this function